Feature/add lom commands

hrgbcxd opened the pull request:
add possibility to send lom commands

https://github.com/micromdm/micromdm/issues/770

hrgbcxd committed:
add lom commands DeviceRequest and SetupRequest

hrgbcxd committed:
init

hrgbcxd committed:
use underscores request_list

hrgbcxd committed:
add missing underscore changes

jessepeterson commented on go.mod:
```diff
@@ -21,7 +21,7 @@ require (
 	golang.org/x/crypto v0.0.0-20211108221036-ceb1ce70b4fa
 	golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
-	google.golang.org/protobuf v1.27.1
+	google.golang.org/protobuf v1.28.0
```
Could you run `go mod tidy`?

hrgbcxd replied:
[run go mod tidy](https://github.com/micromdm/micromdm/pull/839/commits/551b401dad89259f14760a83f0c7d772a84b7026)

ran go mod tidy

jessepeterson commented on mdm/internal/checkinproto/checkin.pb.go:
```diff
@@ -1,7 +1,7 @@
-// 	protoc-gen-go v1.26.0
-// 	protoc        v3.18.1
+// 	protoc-gen-go v1.28.0
+// 	protoc        v3.15.8
```
Go protobuf was updated but it looks like you're running an older version of protoc. Would you mind upgrading that?

hrgbcxd replied:
[upgrade protoc to 3.21.2](https://github.com/micromdm/micromdm/pull/839/commits/2dc7b8d5c51b7a7f6d6f06b778c2ba069f045c33)

upgraded to latest protoc

jessepeterson reviewed (changes requested):
Overall this looks okay in my quick review! What I'd love to see is an end-to-end test in `mdm_command_test.go` though. Could you provide that (as well as the other mentioned things).

jessepeterson commented:
Also - adding some tools scripts would be useful, too. :)

hrgbcxd committed:
run go mod tidy

hrgbcxd committed:
upgrade protoc to 3.21.2

hrgbcxd committed:
add lom_reset to tools scripts

hrgbcxd committed:
add end2end test for LOMDeviceRequest

hrgbcxd commented:
[add lom_reset to tools scripts](https://github.com/micromdm/micromdm/pull/839/commits/c762bd9ce7df47068d685e5d75a5503ce81e7182)

added shell script as example call.

[add end2end test for LOMDeviceRequest](https://github.com/micromdm/micromdm/pull/839/commits/452fa1678e2aa8d349447b7a230d85c688a90f78)

added e2e test case for LOMDeviceRequest

I can run an actual test tomorrow if all this still works, because in [add missing underscore changes](https://github.com/micromdm/micromdm/pull/839/commits/6156c3eaae336d59f840d9772d5cb741f84eff4f) I changed a generated file which now got changed back to camel case.

hrgbcxd committed:
Merge branch 'main' into feature/add_lom_commands

jessepeterson reviewed (approved):
Looks like a bunch of non-related protobufs were ran, too. THat's fun, but unnecessary. :)

This lgtm. Just fix the linter and we can merge. :)

hrgbcxd committed:
formatted code

hrgbcxd committed:
Merge branch 'feature/add_lom_commands' of https://github.com/hrgbcxd/micromdm into feature/add_lom_commands

hrgbcxd committed:
rename primary_id_v6_address_list to primary_ip_v6_address_list

twentyNoise commented:
@hrgbcxd

Hi, I'm trying to get LOM working with MicroMDM and saw your comment that "the lom reset works." Could you share what hardware you tested on? Specifically did it work on a Mac Pro 2019 (T2) or Mac mini? Any special certificate or network configuration? I have extensive logs and am trying to determine if this is a firmware bug. Thanks!
